src/thorlabs_mff_fastcs/controllers.py

```diff
@@ -58,6 +61,56 @@ def read_hardware_v(self, response: bytes) -> int:
 protocol = ThorlabsAPTProtocol()
 
 
+@dataclass
+class ResponseCache:
+    _last_update: datetime | None = None
+    _response: str | None = None
+    _update_event: asyncio.Event = asyncio.Event()
+
+    def __post_init__(self) -> None:
+        self._update_event.set()
+
+    def check_expired(self, time_step: float) -> bool:
+        if self._last_update is None:
+            return True
+        delta_t = datetime.now() - self._last_update
+        return delta_t.total_seconds() > time_step
+
+    def update_response(self, response: str) -> None:
+        self._response = response
+        self._last_update = datetime.now()
```

Comment:
These two methods should be "private" now as they are only called internally.

Reply:
Agree